tiledb/common/platform.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file   tiledb/common/platform.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Platform/machine config of the TileDB library.
+ */
+
```

Comment:
@ihnorton I'll approve this PR but before merging, could you confirm this platform.h is actually what we want moving forward?

Reply:
I'm reading the discussion in slack, per Eric, and think we will probably have a platform library in common, however it will be quite different from this one. I personally think this is fine for now because it serves the feature it is supposed to currently and the platform library is being implemented this iteration.

Reply:
We may end up using an external platform-detection library at some point, but, even if we do that, we'll want to wrap it in our own header. So starting here seems fine by me.

What we don't really want is having a platform-detection library in most source files, but that's independent of the present issue.

Reply:
Moved to https://github.com/TileDB-Inc/TileDB/pull/3420